### DISCUSSION TITLE
When using Discovery Server, a super client cannot see another client in the same domain

### DISCUSSION THREAD

**Discussion by MMarcus95** [eProsima/Fast-DDS#3840] [Support]
Hello,

I'm using the Discovery Server and I would like to allow a Domain Participant (SUPER_CLIENT) to discover all the other clients connected to the server, even if they do not have any Writer or Reader.

However it seems that a SUPER_CLIENT _A_ can discover a CLIENT _B_ , inside the same domain of _A_, if and only if _B_ has at least one Writer or Reader.

Can you explain me why and if there is a solution to discover a CLIENT that does not have any Writer/Reader?

Thanks!

**Comment by JLBuenoLopez** [comment]
Hi  @MMarcus95 

I have issued the following PR to Fast DDS Docs:

* eProsima/Fast-DDS-docs#555

What you are describing is the expected behavior, though it was incorrectly documented. The PR clarifies the expected behavior. 

Using the Discovery Server mechanism, the only DomainParticipant which will discover another DomainParticipant without endpoints will be if it has been configured as a *Server*.

**Reply by MMarcus95** [reply]
Ok thanks for the reply! I'm therefore going to configure my SUPER_CLIENT as redundant server.
